Source/WarpX.cpp

```diff
@@ -1176,52 +1178,57 @@ WarpX::ReadParameters ()
+
         if (use_default_v_comoving == true && boosted_frame == true)
         {
             m_v_comoving[2] = -std::sqrt(1._rt - 1._rt / (gamma_boost * gamma_boost));
         }
-        else if (use_default_v_comoving == true && boosted_frame == false)
-        {
-            amrex::Abort("psatd.use_default_v_comoving = 1 can be used only if warpx.gamma_boost is also set");
-        }
         else
         {
             queryArrWithParser(pp_psatd, "v_comoving", m_v_comoving, 0, 3);
         }
 
-        // Galilean and comoving algorithms should not be used together
-        if (m_v_galilean[0] != 0. || m_v_galilean[1] != 0. || m_v_galilean[2] != 0.)
-        {
-            if (m_v_comoving[0] != 0. || m_v_comoving[1] != 0. || m_v_comoving[2] != 0.)
-            {
-                amrex::Abort("Galilean and comoving algorithms should not be used together");
-            }
-        }
-
         // Scale the Galilean/comoving velocity by the speed of light
-        for (int i=0; i<3; i++) m_v_galilean[i] *= PhysConst::c;
-        for (int i=0; i<3; i++) m_v_comoving[i] *= PhysConst::c;
+        for (auto& vv : m_v_galilean) vv*= PhysConst::c;
+        for (auto& vv : m_v_comoving) vv*= PhysConst::c;
+
+        const auto v_galilean_is_zero =
+            std::all_of(m_v_galilean.begin(), m_v_galilean.end(),
+            [](const auto& val){return val == 0.;});
+
+        const auto v_comoving_is_zero =
+            std::all_of(m_v_comoving.begin(), m_v_comoving.end(),
+            [](const auto& val){return val == 0.;});
+
+
+        // Galilean and comoving algorithms should not be used together
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+            v_galilean_is_zero || v_comoving_is_zero,
+            "Galilean and comoving algorithms should not be used together"
+        );
 
         // The comoving PSATD algorithm is not implemented nor tested with Esirkepov current deposition
-        if (current_deposition_algo == CurrentDepositionAlgo::Esirkepov) {
-            if (m_v_comoving[0] != 0. || m_v_comoving[1] != 0. || m_v_comoving[2] != 0.) {
-                amrex::Abort("Esirkepov current deposition cannot be used with the comoving PSATD algorithm");
-            }
-        }
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+            (current_deposition_algo != CurrentDepositionAlgo::Esirkepov) ||
+            v_comoving_is_zero,
+            "Esirkepov current deposition cannot be used with the comoving PSATD algorithm"
+        );
 
-        if (current_deposition_algo == CurrentDepositionAlgo::Vay) {
-            if (m_v_galilean[0] != 0. || m_v_galilean[1] != 0. || m_v_galilean[2] != 0.) {
-                amrex::Abort("Vay current deposition not implemented for Galilean algorithms");
-            }
-        }
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(
+            (current_deposition_algo != CurrentDepositionAlgo::Vay) ||
+            v_galilean_is_zero,
+            "Vay current deposition not implemented for Galilean algorithms"
```

Comment:
If `current_deposition_algo` is Vay and `v_galilean_is_zero` is True this will not fail. I think it should read:
```suggestion
            (current_deposition_algo == CurrentDepositionAlgo::Vay) ^
            v_galilean_is_zero,
            "Vay current deposition not implemented for Galilean algorithms"
```

Reply:
I think that this is what the previous code did, actually.
With 
```
  if (current_deposition_algo == CurrentDepositionAlgo::Vay) {
            if (m_v_galilean[0] != 0. || m_v_galilean[1] != 0. || m_v_galilean[2] != 0.) {
                //ABORT
            }
}
```
if `current_deposition_algo` is `Vay` the first `if` clause is True. But if `v_galilean_is_zero` is also True, this means that  all the components of `m_v_galilean` are zero. Therefore the second `if` clause would be False and we would **not** abort. 
This is exactly what the assert with `(current_deposition_algo != CurrentDepositionAlgo::Vay) || v_galilean_is_zero` does, I think...